main.ml

```diff
@@ -250,24 +252,23 @@ module Datapath_plugins = struct
 end
 
 let vdi_of_volume x =
-  let ty = match List.Assoc.find x.Xapi_storage.Volume.Types.keys _vdi_type_key ~equal:String.equal with
-    | None -> "";
-    | Some v -> v in
-  let snapshot_time = match List.Assoc.find x.Xapi_storage.Volume.Types.keys _snapshot_time_key ~equal:String.equal with
-    | None -> "19700101T00:00:00Z"
-    | Some snapshot_time -> snapshot_time
+  let find key ~default ~of_string =
+    match List.Assoc.find x.Xapi_storage.Volume.Types.keys key ~equal:String.equal with
+    | None -> default;
```

Comment:
Bit confused, why is there a semicolon here?

Reply:
Probably a typo